### PR TITLE
IC-1078: Link find and refer journey

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

- Adds new PP dashboard placeholder page to start PP referral journey, with a button taking them to the Find interventions search results page.
- Passes selected intervention ID to Refer journey, replacing hardcoded one.
- Updates referral start page routes to take Intervention ID in params - `/interventions/:id/refer`
- Updates copy on User CRN entry page
- Removes referral id from Refer form

## What is the intent behind these changes?

To link up the Find and Refer user journeys.

### Placeholder PP start journey
image

### User CRN lookup
image

### CODE REVIEW

File: server/views/referrals/start.njk [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
       <p class="govuk-body-m">You can make a new referral here:</p>
 
-      <form action="/referrals/start" method="POST">
+      <form action="/intervention/{{presenter.interventionId}}/refer" method="POST">
```
Comment (lawrence-forooghian) [inline server/views/referrals/start.njk:21]: I'd say it's better just to expose this whole href on the presenter.

**Review by lawrence-forooghian — COMMENTED** [review]
Took a quick look, seems sensible, will look again when out of draft 🙂

File: server/routes/referrals/referralFormPresenter.ts [hunk 2]
```diff
@@ -3,11 +3,6 @@ import { DraftReferral } from '../../services/interventionsService'
 export default class ReferralFormPresenter {
   constructor(private readonly referral: DraftReferral, private readonly serviceCategoryName: string) {}
 
-  // This is just temporary, will remove it once we get rid of the referral ID output in the template
```
Comment (lawrence-forooghian) [inline server/routes/referrals/referralFormPresenter.ts:6]: 🎉

**Review by lawrence-forooghian — COMMENTED** [review]
Cool, looks great! A few minor comments.

File: server/routes/referrals/referralStartPresenter.test.ts [hunk 3]
```diff
@@ -35,4 +37,12 @@ describe('ReferralStartPresenter', () => {
       ])
     })
   })
+
+  describe('hrefStartReferral', () => {
+    it('returns a link including the intervention ID to start the referral', () => {
+      const presenter = new ReferralStartPresenter(referrals, interventionId)
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/routes/referrals/referralStartPresenter.test.ts]: Good point. The passing of referrals was actually removed in a later commit, so as long as you're happy to leave this in the commit history, I'll just take this on board for the future!

[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply server/routes/referrals/referralStartPresenter.test.ts]: 👍

File: server/routes/referrals/referralStartPresenter.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ import CalendarDay from '../../utils/calendarDay'
 export default class ReferralStartPresenter {
   constructor(
     private readonly draftReferrals: DraftReferral[],
+    readonly interventionId: string,
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/routes/referrals/referralStartPresenter.ts]: Ah this was a leftover from before I added the `href` method. Have made this private.

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**